spec/models/flag_spec.rb

```diff
@@ -1,7 +1,11 @@
 # frozen_string_literal: true
 
 RSpec.describe Flag, type: :model do
-  before { Flag.reset_flag_settings! }
+  around do |test|
+    Flag.reset_flag_settings!
```

Comment:
Usually, we should only be resetting state after a test has run. If we need to reset state before a test has run, does it mean other tests are leaking state which affects this. test?